src/transformers/models/beit/modeling_beit.py

```diff
@@ -142,14 +142,11 @@ class BeitEmbeddings(nn.Module):
 
     """
 
-    def __init__(self, config):
+    def __init__(self, config, use_mask_token=False):
         super().__init__()
 
         self.cls_token = nn.Parameter(torch.zeros(1, 1, config.hidden_size))
-        if config.use_mask_token:
-            self.mask_token = nn.Parameter(torch.zeros(1, 1, config.hidden_size))
-        else:
-            self.mask_token = None
+        self.mask_token = nn.Parameter(torch.zeros(1, 1, config.hidden_size)) if use_mask_token else None
```

Comment:
This is breaking since if there is a model on the Hub with a checkpoint that has `use_mask_token=True` in its config, but is then instantiated in a class that doesn't send `use_mask_token=True` to this function, we will have a mistmatch in parameters and the load will crash.

We need to have something that is more backward compatible.